File: test-fixtures/cross-language/python/src/error_swallow.py
"""Error-swallow: bare except: pass blocks"""


def swallow_function(path: str) -> None:
    try:
        with open(path) as f:
            return f.read()
    except:
        pass


def parse_silent(input_str: str) -> int:
    try:
        return int(input_str)
    except:
        return 0


def fetch_silent(url: str) -> None:
    try:
        import urllib.request
        urllib.request.urlopen(url)
    except:
        pass

File: test-fixtures/cross-language/rust/src/error_swallow.rs
// Error-swallow: ignoring errors with if let Err(_)
use std::fs::File;
use std::io::Read;

pub fn read_file_silent(path: &str) {
    if let Err(_) = File::open(path) {
        // silently ignore
    }
}

pub fn parse_number_silent(input: &str) -> i32 {
    if let Err(_) = input.parse::<i32>() {
        return 0;
    }
    42
}

pub fn fetch_data_silent(url: &str) {
    if let Err(_) = reqwest::blocking::get(url) {
        // do nothing
    }
}
